src/lib/storage/segment_iterables/any_segment_iterable.cpp

```diff
@@ -11,10 +11,12 @@ AnySegmentIterable<T> CreateAnySegmentIterable<T>::create(const BaseSegment& bas
   auto any_segment_iterable = std::optional<AnySegmentIterable<T>>{};
 
   resolve_segment_type<T>(base_segment, [&](const auto& segment) {
-    const auto actual_iterable = create_iterable_from_segment<T>(segment);
+    const auto actual_iterable = create_iterable_from_segment<T, false>(segment);
```

Comment:
I'm being annoying if I point out that literal boolean parameters are not readable on the calling side?

Reply:
This is literally the only caller of that method that sets the second template parameter. That is unlikely to change, so adding another type just for that seems like overkill. Also it would complicate things on the receiver side, where instead of simply writing

```
template <typename T, bool EraseType = HYRISE_DEBUG>
auto create_iterable_from_segment(const ValueSegment<T>& segment) {
  if constexpr (EraseType) {
```

you would write (untested) 

```
template <typename T, EraseType = EraseType::Yes>
auto create_iterable_from_segment(const ValueSegment<T>& segment) {
  if constexpr (EraseType == EraseType::Yes && HYRISE_DEBUG) {
```

While I started to appreciate seeing `UseMvcc::Yes` instead of `true`, I think we should decide on a case-to-case basis if this pattern really improves readability.

Reply:
kk